### PR TITLE
Ignore querier error in swap simulations

### PR DESCRIPTION
**Pull request by lubkoll**

### CODE REVIEW

File: smart-contracts/contracts/dex-router-osmosis/src/contract.rs [hunk 1]
```diff
@@ -351,8 +357,13 @@ pub fn simulate_swaps(
     path: Vec<SwapAmountInRoute>,
 ) -> Result<Uint128, ContractError> {
     let querier = PoolmanagerQuerier::new(&deps.querier);
-    let response = querier.estimate_swap_exact_amount_in(0, offer.to_string(), path)?;
-    Ok(Uint128::from_str(&response.token_out_amount)?)
+    let response = querier.estimate_swap_exact_amount_in(0, offer.to_string(), path);
```
Comment (magiodev) [inline smart-contracts/contracts/dex-router-osmosis/src/contract.rs:360]: Please leave a comment about the hardcoded pool_id 0 which could be no-sense for a non conteztualized reader.

File: smart-contracts/contracts/dex-router-osmosis/src/error.rs [hunk 2]
```diff
@@ -31,6 +31,9 @@ pub enum ContractError {
 
     #[error("Can't set empty path.")]
     EmptyPath {},
+
+    #[error("Not enough liquidity for swap from {offer:?} to {ask_denom:?}")]
+    NotEnoughLiquidity { offer: Coin, ask_denom: String },
```
Comment (magiodev) [inline smart-contracts/contracts/dex-router-osmosis/src/error.rs:36]: Is that error only about liquidity issues? May it be for another reason so would make more sense a generic SwapError from asset to asset?

**Review by magiodev — COMMENTED** [review]
Lgtm! Minor comments left

File: smart-contracts/contracts/dex-router-osmosis/src/contract.rs (hunk 1 above)
Comment (lubkoll) [reply smart-contracts/contracts/dex-router-osmosis/src/contract.rs:361]: done

File: smart-contracts/contracts/dex-router-osmosis/src/error.rs (hunk 2 above)
Comment (lubkoll) [reply smart-contracts/contracts/dex-router-osmosis/src/error.rs]: done, gave the error a more generic name, other possible causes are
- osmosis messed up (though we shouldn't care about that in our implementation)
- a pool that is used in a path got removed

**Review by magiodev — APPROVED** [review]

**Pull request merged**